### PR TITLE
wip: Cluster shutdown race

### PR DESCRIPTION
**Pull request by jmarantz**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description: Experimental changes to help debug https://github.com/envoyproxy/envoy/issues/9448
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by mattklein123** [comment]
@jmarantz we can close this one right?

File: source/common/stats/thread_local_store.cc [hunk 1]
```diff
@@ -242,14 +245,21 @@ void ThreadLocalStoreImpl::releaseScopeCrossThread(ScopeImpl* scope) {
   }
 }
 
+void ThreadLocalStoreImpl::TlsCache::eraseScope(uint64_t scope_id) { scope_cache_.erase(scope_id); }
```
Comment (jmarantz) [inline source/common/stats/thread_local_store.cc]: @mattklein123 RE closing this PR -- we could, but I was wondering what you thought about submitting the changes around this comment which I think make it easier to debug if there's an issue in this code in the future.

Comment (mattklein123) [reply source/common/stats/thread_local_store.cc]: I'm not sure what changes you are referring to. Can you maybe open a fresh PR?

Comment (jmarantz) [reply source/common/stats/thread_local_store.cc]: I think we've now done most of what I wanted here. closing.

**Pull request closed without merging**